lib/ansible/modules/cloud/amazon/lambda_facts.py

```diff
@@ -155,7 +158,7 @@ def alias_details(client, module):
             if e.response['Error']['Code'] == 'ResourceNotFoundException':
                 lambda_facts.update(aliases=[])
             else:
-                module.fail_json(msg='Unable to get {0} aliases, error: {1}'.format(function_name, e))
+                module.fail_json_aws(e, msg="Trying to get aliases")
     else:
         module.fail_json(msg='Parameter function_name required for query=aliases.')
```

Comment:
I'm not sure about this.  The two functions seem to be used in very different cases;  fail_json seems to happen when the use has done something wrong, like not providing the region, which the ansible module understands best and can explain without giving reference to the code.  In these cases I think it's probably better when we give a simple error ("please provide the region parameter") and don't give a traceback. 

fail_json_aws mostly gets called for failures that happen deep inside botocore logic and the automatic  provinding of a traceback really makes sense.  Especially it should end up good at capturing errors conditions that the module author never realised they had such as _socket.error_.